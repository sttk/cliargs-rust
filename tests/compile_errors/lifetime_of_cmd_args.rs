use cliargs::Cmd;

fn main() {
    fn returns_one_of_cmd_args() -> &'static str {
        let mut cmd = cliargs::Cmd::with_strings(["/path/to/app".to_string(), "foo".to_string()]);
        cmd.parse().unwrap();

        let args = cmd.args();
        let arg1 = args[0];
        println!("command args (within the scope = {arg1:?}");

        arg1
    }

    let arg1 = returns_one_of_cmd_args();
    println!("command args (out of the scope) = {arg1:?}");
}

