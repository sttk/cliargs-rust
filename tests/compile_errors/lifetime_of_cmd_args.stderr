warning: unused import: `cliargs::Cmd`
 --> tests/compile_errors/lifetime_of_cmd_args.rs:1:5
  |
1 | use cliargs::Cmd;
  |     ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0515]: cannot return value referencing local variable `cmd`
  --> tests/compile_errors/lifetime_of_cmd_args.rs:12:9
   |
8  |         let args = cmd.args();
   |                    --- `cmd` is borrowed here
...
12 |         arg1
   |         ^^^^ returns a value referencing data owned by the current function
